src/extensions/sidebar/sidebarext_hooks.cpp

```diff
@@ -26,183 +26,2029 @@
+                {
+                    ready_sw = true;
+                    break;
+                }
+            }
+
+            if (ready_sw && !SidebarExtension->TabButtons[ID].IsFlashing)
+                SidebarExtension->TabButtons[ID].Start_Flashing();
+            else if (!ready_sw && SidebarExtension->TabButtons[ID].IsFlashing)
+                SidebarExtension->TabButtons[ID].Stop_Flashing();
+        }
+    }
+    else
+    {
+        if (SidebarExtension->TabButtons[ID].Is_Enabled())
+            SidebarExtension->TabButtons[ID].Disable();
+    }
+}
+
+
+/**
+ *  Fake function to patch calls to redraw a specific vanilla strip.
+ *
+ *  @author: ZivDero
+ */
+void StripClassFake::_Fake_Flag_To_Redraw_Special()
+{
+    SidebarExtension->Get_Tab(RTTI_SPECIAL).Flag_To_Redraw();
+}
+
+
+/**
+ *  Fake function to patch calls to redraw a specific vanilla strip.
+ *
+ *  @author: ZivDero
+ */
+void StripClassFake::_Fake_Flag_To_Redraw_Current()
+{
+    SidebarExtension->Current_Tab().Flag_To_Redraw();
+}
+
+
+/**
+ *  Patch in GadgetClass::Input to handle hover effects for SelectClass.
+ *
+ *  @author: ZivDero
+ */
+DECLARE_PATCH(_GadgetClass_Input_Mouse_Enter_Leave)
+{
+    GET_REGISTER_STATIC(int, key, eax);
+    GET_REGISTER_STATIC(int, mousex, ebp);
+    GET_REGISTER_STATIC(int, mousey, ebx);
+    GET_REGISTER_STATIC(unsigned, flags, edi);
+    GET_REGISTER_STATIC(GadgetClass*, this_ptr, esi);
+
+    _asm push eax
+    _asm push edx
+    SidebarClassExtension::ViniferaSelectClass::Check_Hover(this_ptr, mousex, mousey);
+    _asm pop edx
+    _asm pop eax
+
+    // Stolen code
+
+    /*
+    **	Set the mouse button state flags. These will be passed to the individual
+    **	buttons so that they can determine what action to perform (if any).
+    */
+    flags = 0;
+    if (key)
+    {
+        if (key == KN_LMOUSE)
+            flags |= GadgetClass::LEFTPRESS;
+        
+        if (key == KN_RMOUSE)
+            flags |= GadgetClass::RIGHTPRESS;
+        
+        if (key == (KN_LMOUSE | KN_RLSE_BIT))
+            flags |= GadgetClass::LEFTRELEASE;
+        
+        if (key == (KN_RMOUSE | KN_RLSE_BIT))
+            flags |= GadgetClass::RIGHTRELEASE;
+
+        /*
+        **	If the mouse wasn't responsible for this key code, then it must be from
+        **	the keyboard. Flag this fact.
+        */
+        if (!flags)
+            flags |= GadgetClass::KEYBOARD;
+
+        _asm mov edi, flags
+        JMP_REG(ecx, 0x004A9F7F);
+    }
+
+    _asm mov edi, flags
+    JMP_REG(ecx, 0x004A9F4D);
+}
+
+
+/**
+ *  Moves the power bar to accomodate for the taller SIDE1.SHP.
+ *
+ *  @author: ZivDero
+ */
+DECLARE_PATCH(_PowerClass_Draw_It_Move_Power_Bar)
+{
+    static int y;
+    y = SidebarRect.Y + SidebarClassExtension::COLUMN_Y - 1;
+    _asm mov esi, y
+
+    static int max_visible;
+    max_visible = SidebarClassExtension::Max_Visible(true);
+    _asm mov eax, max_visible
+    _asm mov ecx, max_visible
```

Comment:
So far all Vinifera code has included braces around `_asm` blocks. Maybe we should stick to that style?